controllers/controllers/services/cfservicebinding_controller.go

```diff
@@ -36,21 +36,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fake -fake-name VCAPServicesSecretBuilder . VCAPServicesSecretBuilder
+type VCAPServicesSecretBuilder interface {
+	BuildVCAPServicesEnvValue(context.Context, *korifiv1alpha1.CFApp) (string, error)
+}
+
 // CFServiceBindingReconciler reconciles a CFServiceBinding object
 type CFServiceBindingReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
-	Log    logr.Logger
+	Scheme  *runtime.Scheme
+	Log     logr.Logger
+	Builder VCAPServicesSecretBuilder
 }
 
-func NewCFServiceBindingReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger) *CFServiceBindingReconciler {
-	return &CFServiceBindingReconciler{Client: client, Scheme: scheme, Log: log}
+func NewCFServiceBindingReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger, builder VCAPServicesSecretBuilder) *CFServiceBindingReconciler {
```

Comment:
That's a good call. We see this pattern across multiple controllers so we created a chore to address it here: https://github.com/cloudfoundry/korifi/issues/1238
If it can't be addressed generically, we'll create a chore to do it specifically for this controller.